### PR TITLE
Use Mu DevOps

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Updates this repo to use mu_devops for Azure Pipeline definitions.

In order to centralize definitions and avoid build churn in individual Mu repos,
some changes are made to allow build administrators to quickly adjust settings:

- Toolchain, VM image, and architecture are controlled by build administrators through centralized
   variable groups
- CI triggers, CI schedules, and PR triggers are controlled by build administrators within the
  individual pipeline UI settings

- [ ] Breaking change?
  - Will this change break pre-existing builds or functionality without action being taken?
  **No**

## How This Was Tested

Verified in test pipelines.

## Integration Instructions

N/A - No developer integration required

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

**Comment by makubacki** [comment]
Note: This will not pass PR gates until I actually update the real pipelines.

Test runs with these changes are here:

- [[TEST] Mu DevOps - CI - GCC5](https://dev.azure.com/projectmu/mu/_build/results?buildId=27964&view=results)
- [[TEST] Mu DevOps - CI - Windows VS](https://dev.azure.com/projectmu/mu/_build/results?buildId=27965&view=results)
- [[TEST] Mu DevOps - PR - GCC5](https://dev.azure.com/projectmu/mu/_build/results?buildId=27966&view=results)
- [[TEST] Mu DevOps - PR - Windows VS](https://dev.azure.com/projectmu/mu/_build/results?buildId=27976&view=results)

File: .azurepipelines/BaseTools-Build-For-Publication.yml [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 # Copyright (C) Microsoft Corporation
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 ##
+
+resources:
+  repositories:
+    - repository: mu_devops
+      type: github
+      endpoint: makubacki
+      name: makubacki/mu_devops
```
Comment (cfernald) [inline .azurepipelines/BaseTools-Build-For-Publication.yml]: friendly reminder to set this make to the official before merging

File: .azurepipelines/BaseTools-Build-For-Publication.yml [hunk 2]
```diff
@@ -97,10 +105,10 @@ jobs:
         filename: stuart_update
         arguments: -c .pytool/CISettings.py
 
-    - template: templates/basetools-build-steps.yml
+    - template: Steps/BuildBaseTools.yml@mu_devops
       parameters:
-        tool_chain_tag: 'VS2019'
         extra_parameters: '--skip_path_env'
+        tool_chain_tag: 'VS2019'
```
Comment (apop5) [inline .azurepipelines/BaseTools-Build-For-Publication.yml:111]: is the 2019 intentional for base tools build?

**Review by apop5 — APPROVED** [review]

File: .azurepipelines/BaseTools-Build-For-Publication.yml (hunk 2 above)
Comment (makubacki) [reply .azurepipelines/BaseTools-Build-For-Publication.yml:111]: I just left it the way it was. This file needs additional follow up, but I wanted to save that for the future given the other work needed at the moment.

File: .azurepipelines/BaseTools-Build-For-Publication.yml (hunk 1 above)
Comment (makubacki) [reply .azurepipelines/BaseTools-Build-For-Publication.yml]: Thanks, I updated that.

**Review by kuqin12 — APPROVED** [review]

**Review by kenlautner — APPROVED** [review]

**Pull request merged**